### DISCUSSION TITLE
compare Azure and Amazon S3 data storage

### DISCUSSION THREAD

**Discussion by bayerfj** [Drexel-UHC/analytics-corner#6] [Infrastructure]
### Name

Jody Bayer

### Title

Project Manager

### Department

Urban Health Collaborative

### Request type

technical/ingratstructure consult for grant submission

### Request description

Compare Azure and S3 and understand advantages and limitations of both for grant submission. Size of data is unknown and usage but will be large. Users will be internal and external to drexel.  Nora Lee will provide more information on the type of data.

### Notes

> I’m working with Ana on a grant application that will require a platform like Azure or Amazon S3 and I wanted to learn more about the differences between the two (pros/cons).    It’s impossible to decide based on our needs because we will not know how much data we’ll eventually have, what types of data, and what access will look like.  So I was hoping to hear about your and Jody’s experiences with Azure and s3. 

_from Nora_

**Comment by jannettim** [comment]
I don't have a lot of experience with AWS. So I don't know how much I can help with a comparison. But if I have some idea of the type of data and usage I can see if I can pull together some information about what azure storage options you might want to consider.

Conceptually azure should be able to store the data and provide options for accessing internally and externally. If the data you are going to be storing is sensitive, we may want to get some people involved who are more experienced with security.

**Comment by bayerfj** [comment]
Hi Matt,
I submitted the ticket for Nora Lee copied along with Ran. Here's the description of the data.

"The grant would provide funding to do many of the things that folks at the UHC are doing for SALURBAL, including providing "high-security-compliant, high-capacity informatics infrastructure suitable for data integration, storage, management, and sharing and ensure that the data meet the FAIR principles: findability, accessibility, interoperability, and reusability."

I can speak about costs for S3. However we only use it for storage of data that doesn't contain PHI.

Jody

Project Manager
Urban Health Collaborative
http://drexel.edu/uhc/

**Comment by jannettim** [comment]
Azure can pretty much handle any type of data. But exactly how much it costs and what storage type is best is going to depend on several factors. So here are some preliminary questions that will help me narrow down what type of storage you need:

- Is the data structure predefined?
- Will the data structure change over time?
- How do you want people to access the data?
- How quick do the response times need to be?

**Comment by ran-codes** [comment]
@jannettim 

>It’s impossible to decide based on our needs because we will not know how much data we’ll eventually have, what types of data, and what access will look like _from Nora's email_

**Comment by jannettim** [comment]
Azure lets you scale up storage so not knowing how big it is is fine for operations (but makes estimating cost difficult). It sounds like there is no defined schema, so you're probably looking at a NoSQL database of some sort then.

So [Azure Cosmos DB ](https://azure.microsoft.com/en-us/services/cosmos-db/#security) or [Azure Blob Storage](https://azure.microsoft.com/en-us/services/storage/blobs/#security) is probably what I'd suggest. Pricing information for Cosmos is [here](https://azure.microsoft.com/en-us/pricing/details/cosmos-db/) and pricing for blob storage is [here](https://azure.microsoft.com/en-us/pricing/details/storage/blobs/).

Though without any idea about what the data flow will be, it will be very difficult to set anything up that will perform well on either AWS or Azure, provide a firm recommendation about which products to use, or an accurate estimate of cost.

**Comment by ran-codes** [comment]
## Welcome @bayerfj 
Thanks for opening your first GitHub issue and welcome to the UHC GitHub 😃! @bayerfj just double checking: is this for the same grant as Leslie's issue #3?

## My two cents
- **AWS vs Azure**: the companies  are very comparable. Both are built to be massively scalable, reliable, performant and cloud services for huge tech companies. **this comparison is more like comparing two brands at a supermarket: Kellogs vs General Mills. Likewise the pricing is going to be very similiar given the same amount of data hosted on both platforms (capitalism for you).** 
-  **Azure vs Amazon S3:** this is not a valid comparison is like comparing Kellogs to a particular oatmal made from General Mills. But Azure block storage is comparable to Amazon S3; its like two brands of big name cereal. Not much difference both pricewise or performance wise.   
- **My intuition for making a recomendation without knowing specific data requirements.** As Nora pointed out we have no idea how much data, how much or what access will look like. Assuming the AWS and Azure are price [comparable ](https://www.msp360.com/resources/blog/amazon-s3-azure-and-google-cloud-prices-compare/)  and both a able to configured to be HIPAA compliant ([Azure doc](https://docs.microsoft.com/en-us/azure/compliance/offerings/offering-hipaa-us), [AWS docs](https://www.dashsdk.com/resource/aws-amazon-s3-hipaa-compliance/#:~:text=Amazon%20S3%20is%20listed%20as,AWS%20Cloud%20Shared%20Responsibility%20Model.)). I think the decision really comes down to experience and what type of infrastructure is already in place at Drexel. Ken Blackney helped set up Azure for the UHC so there is already an IT based infrastructure for Azure; morever, there are a few staff with experience using it for projects.

**My intuition says both would work have similiar performance/cost. But given the infrastructure for Azure and people already using Azure, I think we should build on that foundation for this project. Also, if there is PHI involved we would need to get IT to support to properly configure the storage to be HIPAA compliant.**

@bayerfj Do you know if AWS or S3 storage is used by other groups at Dornsife

**Comment by bayerfj** [comment]
It’s associated with Leslie’s issue #3. Leslie’s issue references multiple data coordinating grants with similar needs. So this is one of them I’m just not sure if it’s the same .

Is there a way I can change this issue to Nora Lee? She’s the one working on the grant and should respond to these questions.

We use Amazon S3 for Gina Lovasi’s data storage. HOWEVER, we don’t use it in the same way Leslie describes in issue #3. Nora confirmed they want to set up something similar to how SALURBAL uses Azure.  She used Azure for the budget in the grant. I think that was probably the best way to go.

Jody

Project Manager
Urban Health Collaborative
http://drexel.edu/uhc/

**Reply by ran-codes** [reply]
@bayerfj when you get an email from GitHub. Just a friendly recomendation to respond to messages via the GitHub website (particularly if its nested like this forum layout). Its easier to navigate and your repsonses won't have the email metadata; this way we keep discussions tidier. You can get to the GitHub website in a email by clicking the "View it on GitHub" button (as seen in the snip below). 

![image](https://user-images.githubusercontent.com/45013044/182644801-4b9a86da-c66c-4de2-9706-b118b701a1b1.png)

**Reply by ran-codes** [reply]
> Is there a way I can change this issue to Nora Lee? She’s the one working on the grant and should respond to these questions.

@bayerfj Nora would need a GitHub account. Sorry for the inconvenience!

**Comment by jannettim** [comment]
> ** Also, if there is PHI involved we would need to get IT to support to properly configure the storage to be HIPAA compliant.**

Drexel's agreement with Microsoft is already intended to be HIPAA compliant as far as I am aware as well. Though it doesn't sound like PHI is involved in this case.

> ** Nora confirmed they want to set up something similar to how SALURBAL uses Azure **

@rl627 you are using a storage account with blob storage right? That's getting served to a web app?

**Reply by ran-codes** [reply]
> Drexel's agreement with Microsoft is already intended to be HIPAA compliant as far as I am aware as well. Though it doesn't sound like PHI is involved in this case.

Yea. we should consult some one maybe Ken Blackney?

>you are using a storage account with blob storage right? That's getting served to a web app?

Yes. Azure blob storage to store all the data; it also acts as an API for the static web app to access data.

**Reply by jannettim** [reply]
I know that back when Dan Yaw was here and the Promise Neighborhood was looking for somewhere to put some data, we did confirm that they can hold PHI. IT forwarded Kelley Traister the business agreement Drexel has with Microsoft confirming it as a HIPAA compliant (as of 6/29/2020). But we can reach out if you feel it is necessary.

And if they want to have a set up similar to salurbal then that's probably what I would suggest, contingent on any clarification about the data requirements. Plus someone at UHC already has experience at setting up that workflow.

**Reply by ran-codes** [reply]
>we did confirm that they can hold PHI. 

Fantastic.

**Comment by bayerfj** [comment]
It's ok. I work on another big data project with Leslie. I'll keep responding since it has implications of that project.


Jody

Project Manager
Urban Health Collaborative
http://drexel.edu/uhc/<https://nam10.safelinks.protection.outlook.com/?url=http%3A%2F%2Fdrexel.edu%2Fuhc%2F&data=04%7C01%7Cfjb47%40drexel.edu%7C6bdf4f822ba64144722e08d99a4173c7%7C3664e6fa47bd45a696708c4f080f8ca6%7C0%7C0%7C637710428185786132%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2FOuJ%2Fu1HXvixSh1jndszN42ssPm%2BQb860weDQwhxg2M%3D&reserved=0>
***@***.***<https://nam10.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.facebook.com%2FDrexelUHC&data=04%7C01%7Cfjb47%40drexel.edu%7C6bdf4f822ba64144722e08d99a4173c7%7C3664e6fa47bd45a696708c4f080f8ca6%7C0%7C0%7C637710428185796088%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=L9KdmDcikoFRcqp6VYOggX9ySa6ilQdmxfE3zhN2DkQ%3D&reserved=0>
***@***.***<https://nam10.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.twitter.com%2FDrexelUHC&data=04%7C01%7Cfjb47%40drexel.edu%7C6bdf4f822ba64144722e08d99a4173c7%7C3664e6fa47bd45a696708c4f080f8ca6%7C0%7C0%7C637710428185796088%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=JC4W%2B4U4RLC2xhNg%2FDBJnJ10HcCD21UJ3JlTQdrfZsg%3D&reserved=0>
***@***.***<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.youtube.com%2Fchannel%2FUC_GiJX3r_O2b_9jzuPdC-sw&data=04%7C01%7Cfjb47%40drexel.edu%7C6bdf4f822ba64144722e08d99a4173c7%7C3664e6fa47bd45a696708c4f080f8ca6%7C0%7C0%7C637710428185806047%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ybzlVSkDdm0rst%2FZ8%2BrZ%2F8yfi%2Ff7wV0mGEZZ0PKDCc4%3D&reserved=0>

Drexel University offices close at 12 PM ET every Friday, June 17-Sept 9, 2022

**Comment by aquistbe** [comment]
I am using AWS for my grant, mostly because it is what is preferred by the UniAndes team that is doing the data processing and analyses. We are using mostly computing instances rather than storage. Getting it set up through Drexel via the NIH STRIDES program was a bit of a hassle. If you do need HIPAA-standard, the Azure option would be better, and having the security folks at Drexel check or set that up would be very helpful than trying to figure it out on AWS without assistance as that can be very complicated. I think either will be adequate for storage purposes, but if you are doing any computing instances for specific tasks, my understanding is that AWS and Google Cloud have a better infrastructure for that.
